components/FilesTable/FilesTable.vue

```diff
@@ -180,6 +201,51 @@ export default {
   mounted: function() {},
 
   methods: {
+    /**
+     * Converts a semver version string to an integer
+     * @param {String} semverVersion
+     */
+    convertSchemaVersionToInteger: function(semverVersion) {
+      // split version number into parts
+      let parts = semverVersion.split('.')
+      // make sure no part is larger than 1023 or else it won't fit
+      // into 32-bit integer
+      parts.forEach(part => {
+        if (part >= 1024) {
+          throw new Error(`Version string invalid, ${part} is too large`)
+        }
+      })
+      let numericVersion = 0
+      // shift all parts either 0, 10, or 20 bits to the left
+      for (let i = 0; i < 3; i++) {
+        numericVersion |= parts[i] << (i * 10)
```

Comment:
now this is something I've never seen before

Reply:
It's interesting I actually found it [here](https://gist.github.com/dislick/914e67444f8f71df3900bd77ccec6091). Thought it was a good solution since these are different than regular number strings.